tests/util.c

```diff
@@ -176,12 +176,10 @@ get_cpu_name (void)
 	search = "hw.model" ;
 	is_pipe = 1 ;
 #else
-	file = NULL ;
+	return name;
 #endif
 
-	if (file == NULL)
-		return name ;
-
+#if defined (__linux__) || defined (__APPLE__) || defined (__FreeBSD__)
```

Comment:
Something's not right there. Why has that `#if` been removed?

Reply:
The problem here is the `pclose` call afterwards which fails to link on Windows. An alternative would be to replace `is_pipe` with `#define IS_PIPE` (+ `#undef IS_PIPE` at the end) and
```
if (is_pipe)
	pclose (file) ;
else
	fclose (file) ;
```
with
```
#ifdef IS_PIPE
	pclose (file) ;
#else
	fclose (file) ;
#endif
```
Do you want me to do that?

Reply:
What problem was this change trying to address?

Reply:
Native Windows (MSVC) doesn't provide `pclose` (though apparently it has [`_pclose`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/pclose)) so it must be removed by the preprocessor.

Reply:
Ok,  how about you just sqash this PR down to a single commit and I'll sort it out,